malaria24/ona/tests/test_tasks.py

```diff
@@ -174,6 +175,177 @@ def test_compile_and_send_digest_email(self):
             mis.email_address]))
         self.assertEqual(len(mail.outbox), 4)
 
+    @responses.activate
+    def test_national_digest_week_variable(self):
+        self.mk_actor(role=MIS,
+                      email_address='manager@example.org')
+        Facility.objects.create(facility_code='342315',
+                                facility_name='Facility 1',
+                                province='Limpopo',
+                                district=u'Example1')
+        self.mk_actor(role=MANAGER_DISTRICT,
+                      email_address='manager@example.org',
+                      facility_code='342315')
+        self.mk_actor(role=MANAGER_PROVINCIAL,
+                      email_address='m2@example.org',
+                      facility_code='342315')
+        self.mk_actor(role=MANAGER_NATIONAL,
+                      email_address='m3@example.org',
+                      facility_code='342315')
+        ehp1 = self.mk_ehp(name='EHP1', email_address='ehp1@example.org')
+        ehp2 = self.mk_ehp(name='EHP2', email_address='ehp2@example.org')
+
+        for i in range(10):
+            case = self.mk_case(gender='female', facility_code='342315')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+
+        for i in range(10):
+            case = self.mk_case(gender='male', facility_code='222222')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+
+        for i in range(10):
+            case = self.mk_case(gender='male', facility_code='333333')
+            case.date_of_birth = datetime.today().strftime("%y%m%d")
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
+            case.save()
+            case.digest = None
+        self.mk_case()
+        compile_and_send_digest_email()
+        NationalDigest.compile_digest().send_digest_email()
+        digest = NationalDigest.compile_digest()
+        context = digest.get_digest_email_data()
+        week = context['week']
+        try:
+            context['week'] is week
+        except ValueError:
+            print "variable not known"
```

Comment:
All that we're doing here is setting `week` to `context['week']`, and then ensuring that `context['week']` is `week`, which is like saying `a = b`, and then testing that `b = a`. It will always be true, so I don't think that there's much point in testing it. 

Maybe we can do something a bit more useful here, like asserting that certain bits of the context are correct. eg. `self.assertEqual(context['week'], '12 2018')`.

The same for the other tests.

Reply:
By the time the code hits ProvincialDigest.compile_digest().send_digest_email() , it fails automatically if there's any other kind of UnboundLocalError where an unknown variable is placed (same goes for NationalDigest or DistrictDigest). Do I still need to make the extra assertion?

Reply:
No, the extra assertions are probably not necessary. You can probably just remove this try/except block.

Reply:
okay